### ISSUE TITLE
bug: Autocomplete does not work when there's new line at the end of prompt

### ISSUE THREAD

**Issue by doganalper** [withfig/fig#2424]
### Checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue
- [X] I have run `fig doctor` in the affected terminal session
- [X] I have run `fig restart` and replicated the issue again

### Operating system

Macos

### Expected behaviour

Display dropdown menu when prompt ends with new line character.

### Actual behaviour

Does not display dropdown menu at all.

### Steps to reproduce

Add `NEW_LINE=$'\n'` at the end of exported prompt string on `.zshrc`.

```zsh
COLOR_DEF=$'\e[0m'
COLOR_DIR=$'\e[38;5;197m'
COLOR_GIT=$'\e[38;5;39m'
NEW_LINE=$'\n'
setopt PROMPT_SUBST
export PROMPT='${COLOR_DIR}%2~ ${COLOR_GIT}$(parse_git_branch)${COLOR_DEF}${NEW_LINE}'

```

**Comment by mschrage** [comment]
<img width="1143" alt="Screenshot 2023-05-01 at 3 42 00 PM" src="https://user-images.githubusercontent.com/3155109/235544037-25b2343a-010a-43aa-b5e9-464effbfe7f6.png">

It looks like there is an off by one error happening somewhere. cc @grant0417

**Comment by chaynabors** [comment]
👋 Howdy @doganalper. A fix for this should be in Fig 2.16.0 which we'll likely release to beta this week.

**Comment by doganalper** [comment]
@chaynabors Great! I loved using fig until came across to this problem. Thanks for taking care of it.
